src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -320,7 +463,233 @@ private void updateWorkOrders()
+        int index = 0;
+        if (pane instanceof Label)
+        {
+            index = Integer.valueOf(((Label) pane).getLabelText());
+        }
+        final boolean trigger = LanguageHandler.format(ON).equals(button.getLabel());
+        final Permissions.Action action = Permissions.Action.values()[index];
+        final Permissions.Rank rank = Permissions.Rank.valueOf(actionsList.getParent().getID().toUpperCase());
+
+        MineColonies.getNetwork().sendToServer(new PermissionsMessage.Permission(townHall.getColony(), PermissionsMessage.MessageType.TOGGLE_PERMISSION, rank, action));
+        townHall.getColony().getPermissions().togglePermission(rank, action);
+
+        if (trigger)
+        {
+            button.setLabel(LanguageHandler.format(OFF));
+        }
+        else
+        {
+            button.setLabel(LanguageHandler.format(ON));
+        }
+    }
+
+    /**
+     * Switch between previous and next page.
+     */
+    private void switchPage(@NotNull final Button button)
+    {
+        if (button.getID().equals(BUTTON_PREV_PAGE_PERM))
+        {
+            findPaneOfTypeByID(VIEW_PERM_PAGES, SwitchView.class).previousView();
+
+            findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setEnabled(false);
+            findPaneOfTypeByID(BUTTON_NEXT_PAGE_PERM, Button.class).setEnabled(true);
+        }
+        else
+        {
+            findPaneOfTypeByID(VIEW_PERM_PAGES, SwitchView.class).nextView();
+
+            findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setEnabled(true);
+            findPaneOfTypeByID(BUTTON_NEXT_PAGE_PERM, Button.class).setEnabled(false);
+        }
+
+        if (findPaneOfTypeByID(VIEW_PERM_PAGES, SwitchView.class).getCurrentView().getID().equals(PERMISSION_VIEW))
+        {
+            findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setEnabled(true);
+            findPaneOfTypeByID(BUTTON_NEXT_PAGE_PERM, Button.class).setEnabled(true);
+
+            fillPermissionList(VIEW_OFFICER);
+        }
+    }
+
+    /**
+     * Fills the permission list in the GUI.
+     */
+    private void fillPermissionList(@NotNull final String category)
+    {
+        actionsList = findPaneOfTypeByID(LIST_ACTIONS + category, ScrollingList.class);
+        actionsList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return Permissions.Action.values().length - IGNORE_INDEX;
+            }
+
+            @Override
+            public void updateElement(final int index, @NotNull final Pane rowPane)
+            {
+                final int actionIndex = index <= IGNORE_INDEX ? index : (index + IGNORE_INDEX);
+                final Permissions.Action action = Permissions.Action.values()[actionIndex];
+                final String name = LanguageHandler.format(KEY_TO_PERMISSIONS + action.toString().toLowerCase());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define the locale to be used in this String operation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1449)